### PR TITLE
DISTMYSQL-421: Remove master-slave terminologies in Orchestrator

### PR DESCRIPTION
**Pull request by kamil-holubicki**
1. Orchestrator calls intercepted by QueryStringProvider
2. Test framework adjusted to be able to use 8.4

Temporary solution:
Hacked dbdeployer added (out of the box it doesn't work with 8.4)

### CODE REVIEW

File: go/golib/sqlutils/sqlutils.go [hunk 1]
```diff
@@ -170,20 +170,35 @@ func (this *RowMap) GetTime(key string) time.Time {
...
 }
 
+func validateQuery(query string, db *sql.DB) {
+	// dev purposes only. Remove this return to call query validator function.
+	return
+
+	knownDBsMutex.RLock()
+	defer func() {
+		knownDBsMutex.RUnlock()
+	}()
```
Comment (inelpandzic) [inline go/golib/sqlutils/sqlutils.go:180]: You can have `defer knownDBsMutex.RUnlock()`

File: go/db/tls.go [hunk 2]
```diff
@@ -57,22 +57,46 @@ func init() {
...
 }
 
+// This validator is for dev purposes only. Call of this validator is
+// disabled in sqlutils.go
+var query_whitelist = []string{
+	"substring_index(host, ':', 1) as slave_hostname",
+}
+
+func (logger SqlUtilsLogger) ValidateQuery(query string) {
+	if !logger.backend_connection {
```
Comment (ademidoff) [inline go/db/tls.go:84]: I'd revert it:
```suggestion
	if logger.backend_connection {
	   return ""
	}
	
	... put the rest of the code
```

File: go/golib/sqlutils/sqlutils.go (hunk 1 above)
Comment (kamil-holubicki) [reply go/golib/sqlutils/sqlutils.go]: done

File: go/inst/instance_dao.go [hunk 3]
```diff
@@ -595,13 +598,14 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
 				// ...
 				// @@gtid_mode only available in Orcale MySQL >= 5.6
 				// Previous version just issued this query brute-force, but I don't like errors being issued where they shouldn't.
-				_ = db.QueryRow("select @@global.gtid_mode, @@global.server_uuid, @@global.gtid_executed, @@global.gtid_purged, @@global.master_info_repository = 'TABLE', @@global.binlog_row_image").Scan(&instance.GTIDMode, &instance.ServerUUID, &instance.ExecutedGtidSet, &instance.GtidPurged, &masterInfoRepositoryOnTable, &instance.BinlogRowImage)
```
Comment (venkatesh-prasad-v) [inline go/inst/instance_dao.go:598]: What happens when `@@global.master_info_repository = FILE`? We just need to make sure that it works properly on 5.7 instances as well which supports file repository.

File: go/inst/query_string_provider.go [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+var queryStringProvider80 = QueryStringProvider{
+	queries: queryStrings80,
+}
+
+var queryStringProvider84 = QueryStringProvider{
+	queries: queryStrings84,
+}
+
+func GetQueryStringProvider(version string) QueryStringProvider {
+	if strings.HasPrefix(version, "8.4") {
```
Comment (venkatesh-prasad-v) [inline go/inst/query_string_provider.go:347]: If not now, in future we will need to normalize version numbers as intergers for comparisions

File: go/inst/instance_dao.go (hunk 3 above)
Comment (venkatesh-prasad-v) [reply go/inst/instance_dao.go:598]: As I see it is already handled.
For 8.0 and below the query is:
`"select @@global.gtid_mode, @@global.server_uuid, @@global.gtid_executed, @@global.gtid_purged, @@global.master_info_repository = 'TABLE', @@global.binlog_row_image",`
For 8.4 the query is:
`"select @@global.gtid_mode, @@global.server_uuid, @@global.gtid_executed, @@global.gtid_purged, 1, @@global.binlog_row_image",`

Please ignore this comment

Comment (kamil-holubicki) [reply go/inst/instance_dao.go:598]: As we spoke on Slack, below 8.4 the query is:
`"select @@global.gtid_mode, @@global.server_uuid, @@global.gtid_executed, @@global.gtid_purged, 1, @@global.binlog_row_image"`
and for 8.4:
`"select @@global.gtid_mode, @@global.server_uuid, @@global.gtid_executed, @@global.gtid_purged, 1, @@global.binlog_row_image"`

File: go/inst/query_string_provider.go (hunk 4 above)
Comment (kamil-holubicki) [reply go/inst/query_string_provider.go:347]: Let's keep it simple for now. We can always do more magic here when needed.

**Pull request merged**